src/meta/meta_turtle.c

```diff
@@ -94,26 +94,34 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
 }
 
 /*
- * __metadata_load_bulk --
- *     Create any bulk-loaded file stubs.
+ * __metadata_load_bulk_max_write_gen --
+ *     Create any bulk-loaded file stubs and aggregate the maximum write gen.
  */
 static int
-__metadata_load_bulk(WT_SESSION_IMPL *session)
+__metadata_load_bulk_max_write_gen(WT_SESSION_IMPL *session, uint64_t *max_write_gen)
 {
+    WT_CKPT ckpt;
     WT_CURSOR *cursor;
     WT_DECL_RET;
     uint32_t allocsize;
     const char *filecfg[] = {WT_CONFIG_BASE(session, file_meta), NULL, NULL};
     const char *key, *value;
     bool exist;
 
+    *max_write_gen = 0;
     /*
      * If a file was being bulk-loaded during the hot backup, it will appear in the metadata file,
      * but the file won't exist. Create on demand.
      */
     WT_RET(__wt_metadata_cursor(session, &cursor));
     while ((ret = cursor->next(cursor)) == 0) {
         WT_ERR(cursor->get_key(cursor, &key));
+        if ((ret = __wt_meta_checkpoint(session, key, NULL, &ckpt) == 0)) {
```

Comment:
Did you think about adding a new function in meta_ckpt.c that finds this value, rather than embedding it in the existing loop here? The way this works is that we open a metadata cursor here, and for each entry in the metadata we are opening another metadata cursor, searching, populating the checkpoint information and returning it here for the one field we need.

I think it'd be neater to add a new `__wt_metadata_get_max_write_gen` - there is precedent for that with `__wt_metadata_set_base_write_gen`

Reply:
Yeah, I thouht about that. In the meeting, sue suggested we can decode the write gen in line to reduce ovethead. I didn't expect it to work like this though. I'll move it outside.